File: frontend/src/types/stats.ts
import type { CategoryCount, LanguageCount } from './common';

// ---- Dashboard Stats ----
export interface PaperStats {
  total: number;
  by_source: Record<string, number>;
  new_today: number;
  new_this_week: number;
}

export interface RepoStats {
  total: number;
  new_today: number;
  new_this_week: number;
}

export interface TagStats {
  total: number;
}

export interface CrawlStats {
  last_arxiv: string | null;
  last_github: string | null;
  total_runs: number;
  success_rate: number;
}

export interface ReportStats {
  daily_count: number;
  weekly_count: number;
}

export interface StatsResponse {
  papers: PaperStats;
  repos: RepoStats;
  tags: TagStats;
  crawls: CrawlStats;
  reports: ReportStats;
  top_categories: CategoryCount[];
  top_languages: LanguageCount[];
}

export interface HealthResponse {
  status: string;
  timestamp: string;
  database: string;
  scheduler: string;
  version: string;
}

File: backend/app/schemas/stats.py
"""System stats Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class PaperStats(BaseModel):
    total: int
    by_source: dict[str, int]
    new_today: int
    new_this_week: int


class RepoStats(BaseModel):
    total: int
    new_today: int
    new_this_week: int


class TagStats(BaseModel):
    total: int


class CrawlStats(BaseModel):
    last_arxiv: Optional[datetime] = None
    last_github: Optional[datetime] = None
    total_runs: int
    success_rate: float


class ReportStats(BaseModel):
    daily_count: int
    weekly_count: int


class CategoryCount(BaseModel):
    name: str
    paper_count: int


class LanguageCount(BaseModel):
    language: str
    repo_count: int


class StatsResponse(BaseModel):
    papers: PaperStats
    repos: RepoStats
    tags: TagStats
    crawls: CrawlStats
    reports: ReportStats
    top_categories: list[CategoryCount]
    top_languages: list[LanguageCount]


class HealthResponse(BaseModel):
    status: str
    timestamp: datetime
    database: str
    scheduler: str
    version: str
